resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```diff
@@ -280,11 +291,118 @@ private[spark] class Client(
+      val secrets = mutable.Buffer[Secret]()
+      driverLaunchSslOptions.keyStore.foreach(store => {
+        val resolvedKeyStoreFile = if (isKeyStoreLocalFile) {
+          if (!store.isFile) {
+            throw new SparkException(s"KeyStore specified at $store is not a file or" +
+              s" does not exist.")
+          }
+          val keyStoreBytes = Files.toByteArray(store)
+          val keyStoreBase64 = Base64.encodeBase64String(keyStoreBytes)
+          sslSecretsMap += (SSL_KEYSTORE_SECRET_NAME -> keyStoreBase64)
+          s"$sslSecretsDirectory/$SSL_KEYSTORE_SECRET_NAME"
+        } else {
+          store.getAbsolutePath
+        }
+        sslEnvs += new EnvVarBuilder()
+          .withName("SPARK_SUBMISSION_KEYSTORE_FILE")
+          .withValue(resolvedKeyStoreFile)
+          .build()
+      })
+      driverLaunchSslOptions.keyStorePassword.foreach(password => {
+        val passwordBase64 = Base64.encodeBase64String(password.getBytes(Charsets.UTF_8))
+        sslSecretsMap += (SSL_KEYSTORE_PASSWORD_SECRET_NAME -> passwordBase64)
+        sslEnvs += new EnvVarBuilder()
+          .withName("SPARK_SUBMISSION_KEYSTORE_PASSWORD_FILE")
+          .withValue(s"$sslSecretsDirectory/$SSL_KEYSTORE_PASSWORD_SECRET_NAME")
+          .build()
+      })
+      driverLaunchSslOptions.keyPassword.foreach(password => {
+        val passwordBase64 = Base64.encodeBase64String(password.getBytes(Charsets.UTF_8))
+        sslSecretsMap += (SSL_KEY_PASSWORD_SECRET_NAME -> passwordBase64)
+        sslEnvs += new EnvVarBuilder()
+          .withName("SPARK_SUBMISSION_KEYSTORE_KEY_PASSWORD_FILE")
+          .withValue(s"$sslSecretsDirectory/$SSL_KEY_PASSWORD_SECRET_NAME")
+          .build()
+      })
+      driverLaunchSslOptions.keyStoreType.foreach(storeType => {
+        sslEnvs += new EnvVarBuilder()
+          .withName("SPARK_SUBMISSION_KEYSTORE_TYPE")
+          .withValue(storeType)
+          .build()
+      })
+      sslEnvs += new EnvVarBuilder()
+        .withName("SPARK_SUBMISSION_USE_SSL")
+        .withValue("true")
+        .build()
+      val sslSecrets = kubernetesClient.secrets().createNew()
+        .withNewMetadata()
+        .withName(sslSecretsName)
+        .endMetadata()
+        .withData(sslSecretsMap.asJava)
+        .withType("Opaque")
+        .done()
+      secrets += sslSecrets
+      val sslVolume = new VolumeBuilder()
+        .withName("spark-submission-server-ssl-secrets")
+        .withNewSecret()
+          .withSecretName(sslSecrets.getMetadata.getName)
+          .endSecret()
+        .build()
+      val sslVolumeMount = new VolumeMountBuilder()
+        .withName("spark-submission-server-ssl-secrets")
+        .withReadOnly(true)
+        .withMountPath(sslSecretsDirectory)
+        .build()
+      (sslEnvs.toArray, Array(sslVolume), Array(sslVolumeMount), secrets.toArray)
+    } else {
+      (Array[EnvVar](), Array[Volume](), Array[VolumeMount](), Array[Secret]())
+    }
```

Comment:
Regarding if-else ordering - if we want to avoid the `return` statement here, then I prefer checking the positive of a conditional when branching with `if... else`.